### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by ozgunozerk**
Upgrades Rust - Subspace - Substrate

Reviewing all at once will make more sense (i will squash merge them anyway).

### CODE REVIEW

File: src-tauri/Cargo.toml [hunk 1]
```diff
@@ -8,35 +8,38 @@ repository = "https://github.com/subspace/subspace-desktop"
...
+sc-consensus-slots = { version = "0.10.0-dev", git = "https://github.com/subspace/substrate", rev = "1399afdcd8ab4d7fad149c96f9cadcaf04b94f86" }
+sc-executor = { version = "0.10.0-dev", git = "https://github.com/subspace/substrate", rev = "1399afdcd8ab4d7fad149c96f9cadcaf04b94f86", features = ["wasmtime"] }
+sc-network = { version = "0.10.0-dev", git = "https://github.com/subspace/substrate", rev = "1399afdcd8ab4d7fad149c96f9cadcaf04b94f86" }
+sc-service = { version = "0.10.0-dev", git = "https://github.com/subspace/substrate", rev = "1399afdcd8ab4d7fad149c96f9cadcaf04b94f86", features = ["wasmtime"] }
+sc-subspace-chain-specs = { git = "https://github.com/subspace/subspace", rev = "d736cf913f445a8e7dccd5c663e71b19212085ee" }
+serde_json = "1.0.82"
+serde = { version = "1.0.139", features = [ "derive" ] }
+sp-core = { version = "6.0.0", git = "https://github.com/subspace/substrate", rev = "1399afdcd8ab4d7fad149c96f9cadcaf04b94f86" }
+sp-panic-handler = { version = "4.0.0", git = "https://github.com/subspace/substrate", rev = "1399afdcd8ab4d7fad149c96f9cadcaf04b94f86" }
```
Comment (nazar-pc) [inline src-tauri/Cargo.toml:34]: Why is `sp-panic-handler` necessary?

File: src-tauri/Cargo.toml [hunk 2]
```diff
@@ -8,35 +8,38 @@ repository = "https://github.com/subspace/subspace-desktop"
...
+subspace-core-primitives = { git = "https://github.com/subspace/subspace", rev = "d736cf913f445a8e7dccd5c663e71b19212085ee" }
+subspace-farmer = { git = "https://github.com/subspace/subspace", rev = "d736cf913f445a8e7dccd5c663e71b19212085ee" }
+subspace-fraud-proof = { git = "https://github.com/subspace/subspace", rev = "d736cf913f445a8e7dccd5c663e71b19212085ee" }
+subspace-networking = { git = "https://github.com/subspace/subspace", rev = "d736cf913f445a8e7dccd5c663e71b19212085ee" }
+subspace-rpc-primitives = { git = "https://github.com/subspace/subspace", rev = "d736cf913f445a8e7dccd5c663e71b19212085ee" }
```
Comment (nazar-pc) [inline src-tauri/Cargo.toml:39]: `subspace-rpc-primitives` seems to be unused too

File: src-tauri/Cargo.toml [hunk 3]
```diff
@@ -8,35 +8,38 @@ repository = "https://github.com/subspace/subspace-desktop"
...
 anyhow = "1.0.44"
-cirrus-runtime = { git = "https://github.com/subspace/subspace", rev = "61c60e35fc03e286bb91fd61f799c0e279dcb351" }
+cirrus-runtime = { git = "https://github.com/subspace/subspace", rev = "d736cf913f445a8e7dccd5c663e71b19212085ee" }
 dirs = "4.0.0"
 dotenv = "0.15.0"
 event-listener-primitives = "2.0.1"
 fdlimit = "0.2.1"
 fs2 = "0.4.3"
 hex = "0.4.3"
 log = "0.4.14"
```
Comment (nazar-pc) [inline src-tauri/Cargo.toml:23]: Do you still need `log` after switching to `tracing`?

File: src-tauri/src/farmer.rs [hunk 4]
```diff
@@ -69,55 +105,89 @@ async fn farm(
...
-            listen_on: vec![],
-            enable_dsn_archiving: false,
-            dsn_sync: false,
+            bootstrap_nodes,
+            enable_dsn_archiving: matches!(archiving, ArchivingFrom::Dsn),
+            enable_dsn_sync: dsn_sync,
             enable_farming: true,
+            relay_server_node,
         },
         plot_size,
```
Comment (nazar-pc) [inline src-tauri/src/farmer.rs:173]: `LegacyMultiPlotsFarm` was reducing input to 92% before, but it is not going it anymore. I think you need to do it here now in order to make code equivalent to the old version.

File: src-tauri/Cargo.toml (hunk 2 above)
Comment (ozgunozerk) [reply src-tauri/Cargo.toml:39]: Yes... Removing :)
I ran `cargo udeps`. It was not able to catch it. Are you using any tool for detecting these? I would love to learn it

Comment (nazar-pc) [reply src-tauri/Cargo.toml:39]: This time I used my :eyes:

**Comment by ozgunozerk** [comment]
Seems like this will do. 
Did not create any bug during launch/runtime. 
Will test tomorrow for block production and rewards (both locally, and also imitating remote). If there are no errors, we can merge it.

File: src-tauri/src/farmer.rs [hunk 5]
```diff
@@ -170,7 +170,7 @@ async fn farm(base_directory: PathBuf, farm_args: FarmingArgs) -> Result<(), any
             enable_farming: true,
             relay_server_node,
         },
-        plot_size,
+        plot_size * 92 / 100, // LegacyMultiPlotsFarm was reducing input to 92% before, but it is not going it anymore. This is to make code equivalent to the old version.
```
Comment (nazar-pc) [inline src-tauri/src/farmer.rs:173]: This in farmer much more readable and clearly states that this is not a final value:
https://github.com/subspace/subspace/blob/a1b9a3991a8db5d598f616d5c3f34cb8c54905ac/crates/subspace-farmer/src/bin/subspace-farmer/utils.rs#L28-L32

**Review by nazar-pc — APPROVED** [review]

**Comment by ozgunozerk** [comment]
Tested on dev chain-spec, it is able farm blocks! 
If everything else is ok, it is ready to merge

**Review by i1i1 — APPROVED** [review]
LGTM

**Review by nazar-pc — APPROVED** [review]

**Pull request merged**